test/functional/mempool_expiry.py

```diff
@@ -26,73 +26,78 @@
 class MempoolExpiryTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
+        self.setup_clean_chain = True
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+    def trigger_mempool_expiry(self):
+        """Broadcasts an unrelated transaction as the expiry of the transactions in the mempool is only checked
+        when a new transaction is added to the mempool."""
+        try:
+            self.wallet.send_self_transfer(from_node=self.nodes[0])
+        except JSONRPCException:
+            pass
 
     def test_transaction_expiry(self, timeout):
-        """Tests that a transaction expires after the expiry timeout and its
-        children are removed as well."""
+        """Tests that a transaction expires after the expiry timeout and its children are removed as well."""
         node = self.nodes[0]
+        self.wallet = MiniWallet(node)
+
+        # Add enough mature utxos to the wallet so that all txs spend confirmed coins.
+        self.wallet.generate(2)
+        node.generate(100)
 
         # Send a parent transaction that will expire.
-        parent_address = node.getnewaddress()
-        parent_txid = node.sendtoaddress(parent_address, 1.0)
+        parent_txid = self.wallet.send_self_transfer(from_node=node)['txid']
+        parent_utxo = self.wallet.get_utxo(txid=parent_txid)
+        independent_utxo = self.wallet.get_utxo()
 
         # Set the mocktime to the arrival time of the parent transaction.
         entry_time = node.getmempoolentry(parent_txid)['time']
         node.setmocktime(entry_time)
 
-        # Create child transaction spending the parent transaction
-        vout = find_vout_for_address(node, parent_txid, parent_address)
-        inputs = [{'txid': parent_txid, 'vout': vout}]
-        outputs = {node.getnewaddress(): 0.99}
-        child_raw = node.createrawtransaction(inputs, outputs)
-        child_signed = node.signrawtransactionwithwallet(child_raw)['hex']
-
-        # Let half of the timeout elapse and broadcast the child transaction.
-        half_expiry_time = entry_time + int(60 * 60 * timeout/2)
+        # Let half of the timeout elapse.
+        half_expiry_time = entry_time + 3600 * timeout // 2
```

Comment:
it looks like you are fixing up style in the same commit as refactors, cleanups and miniwallet changes

It would be easier to follow if the formatting changes, the `trigger_mempool_expiry`, and the miniwallet changes were all in separate commits, where each passed on its own and can be reviewed on its own

Reply:
Ah yeah, I did too much extra stuff in this PR. I'm gonna undo all this unrelated style/refactor stuff. Will undo the string formatting changes as well